src/babel.js

```diff
@@ -15,16 +15,30 @@ const STYLE_COMPONENT_ID = 'styleId'
 const STYLE_COMPONENT_CSS = 'css'
 
 export default function ({types: t}) {
-  const findStyles = children => (
-    children.filter(el => (
-      t.isJSXElement(el) &&
-      el.openingElement.name.name === 'style' &&
-      el.openingElement.attributes.some(attr => (
-        attr.name.name === STYLE_ATTRIBUTE
-      ))
+  const isGlobalEl = el => el.attributes.some(attr => (
+    attr.name.name === GLOBAL_ATTRIBUTE
+  ))
+
+  const isStyledJsx = ({node: el}) => (
+    t.isJSXElement(el) &&
+    el.openingElement.name.name === 'style' &&
+    el.openingElement.attributes.some(attr => (
+      attr.name.name === STYLE_ATTRIBUTE
     ))
   )
 
+  const findStyles = children => {
+    const firstChild = children[0]
+    if (children.length === 1 && !isStyledJsx(firstChild)) {
+      const parent = firstChild.parentPath
+      return (
+        isStyledJsx(parent) && isGlobalEl(parent.node.openingElement) ?
+        [parent.node] : []
+      )
+    }
+    return children.filter(isStyledJsx).map(({node}) => node)
+  }
```

Comment:
I wonder if we can't traverse from parent node instead of `findStyles(path.node.children)` ? It might makes things simpler.